src/nl/hannahsten/texifyidea/inspections/latex/LatexPackageMayNotExistInspection.kt

```diff
@@ -0,0 +1,50 @@
+package nl.hannahsten.texifyidea.inspections.latex
+
+import com.intellij.codeInspection.InspectionManager
+import com.intellij.codeInspection.ProblemDescriptor
+import com.intellij.codeInspection.ProblemHighlightType
+import com.intellij.psi.PsiFile
+import nl.hannahsten.texifyidea.index.LatexDefinitionIndex
+import nl.hannahsten.texifyidea.insight.InsightGroup
+import nl.hannahsten.texifyidea.inspections.TexifyInspectionBase
+import nl.hannahsten.texifyidea.psi.LatexCommands
+import nl.hannahsten.texifyidea.util.Magic
+import nl.hannahsten.texifyidea.util.PackageUtils
+import nl.hannahsten.texifyidea.util.childrenOfType
+import nl.hannahsten.texifyidea.util.projectSearchScope
+
+class LatexPackageMayNotExistInspection : TexifyInspectionBase() {
+    override val inspectionGroup: InsightGroup = InsightGroup.LATEX
+
+    override val inspectionId: String =
+            "PackageMayNotExist"
+
+    override fun getDisplayName(): String {
+        return "Package may not exist"
+    }
+
+    override fun inspectFile(file: PsiFile, manager: InspectionManager, isOntheFly: Boolean): List<ProblemDescriptor> {
+        val descriptors = descriptorList()
+        val ctanPackages = PackageUtils.CTAN_PACKAGE_NAMES
+        val customPackages = LatexDefinitionIndex.getCommandsByName("\\ProvidesPackage", file.project, file.project.projectSearchScope)
```

Comment:
When I have a package `packages/mypackage.sty` and there I have `\ProvidesPackage{chapters/mypackage}[2020/02/04 thomas's Package]`, then include it with `\usepackage{packages/mypackage}` then the inspection cannot find it

Maybe you can add a test for this?

Reply:
Correct. I don't know how this was supposed to find the packages... Oops.